File: dart/data_structures/queue.dart
// queue.dart

class Queue {
  List<dynamic> _items = [];

  bool get isEmpty => _items.isEmpty;

  void enqueue(dynamic item) {
    _items.add(item);
  }

  dynamic dequeue() {
    if (isEmpty) {
      throw StateError("Queue is empty");
    }
    return _items.removeAt(0);
  }

  dynamic front() {
    if (isEmpty) {
      throw StateError("Queue is empty");
    }
    return _items.first;
  }

  void display() {
    print("Queue: $_items");
  }
}

void main() {
  Queue myQueue = Queue();

  myQueue.enqueue(1);
  myQueue.enqueue(2);
  myQueue.enqueue(3);

  print("Original Queue:");
  myQueue.display();

  print("\nDequeue operation result: ${myQueue.dequeue()}");

  print("Queue after Dequeue operation:");
  myQueue.display();
}

File: go/data_structures/queue.go
// queue.go

package main

import "fmt"

type Queue struct {
	items []int
}

func (q *Queue) Enqueue(item int) {
	q.items = append(q.items, item)
}

func (q *Queue) Dequeue() (int, error) {
	if len(q.items) == 0 {
		return 0, fmt.Errorf("queue is empty")
	}

	item := q.items[0]
	q.items = q.items[1:]
	return item, nil
}

func (q *Queue) IsEmpty() bool {
	return len(q.items) == 0
}

func (q *Queue) Size() int {
	return len(q.items)
}

func (q *Queue) Display() {
	fmt.Println("Queue:", q.items)
}

func main() {
	queue := &Queue{}

	queue.Enqueue(1)
	queue.Enqueue(2)
	queue.Enqueue(3)

	fmt.Println("Queue Size:", queue.Size())

	item, err := queue.Dequeue()
	if err == nil {
		fmt.Println("Dequeued Item:", item)
	} else {
		fmt.Println(err)
	}

	fmt.Println("Is Queue Empty:", queue.IsEmpty())
	queue.Display()
}
